### PR TITLE
En_Kanban

### PR DESCRIPTION
**Pull request by petrie911**
Just think, the devs took time to ensure wooden signs had functioning dynamic shadows but not Link.

### CODE REVIEW

File: src/overlays/actors/ovl_En_Kanban/z_en_kanban.c [hunk 1]
```diff
@@ -25,15 +83,777 @@ const ActorInit En_Kanban_InitVars = {
...
+                    } else {
+                        piece->bounceX = (s16)Rand_ZeroFloat(7.0f) + 3;
+                        piece->bounceZ = (s16)Rand_ZeroFloat(7.0f) + 3;
+                    }
+                    piece->spinVel.y = Rand_CenteredFloat(0x1800);
+                    if (Rand_ZeroOne() < 0.5f) {
+                        piece->direction = 1;
+                    } else {
+                        piece->direction = -1;
+                    }
```
Comment (fig02) [inline src/overlays/actors/ovl_En_Kanban/z_en_kanban.c:371]: ternary might look nice here, but its fine either way

File: src/overlays/actors/ovl_En_Kanban/z_en_kanban.c [hunk 2]
```diff
@@ -25,15 +83,777 @@ const ActorInit En_Kanban_InitVars = {
...
+            func_8002E4B4(globalCtx, &this->actor, 10.0f, 10.0f, 50.0f, 4);
+            EnKanban_SetFloorRot(this);
+
+            this->actor.posRot.pos.x = tempX;
+            this->actor.posRot.pos.y = tempY;
+            this->actor.posRot.pos.z = tempZ;
+            this->actor.bgCheckFlags = tempBgFlags;
+            this->actor.waterY = tempWaterY;
+
+            osSyncPrintf("\x1b[m");
```
Comment (fig02) [inline src/overlays/actors/ovl_En_Kanban/z_en_kanban.c:431]: use vt macros

File: src/overlays/actors/ovl_En_Kanban/z_en_kanban.c [hunk 3]
```diff
@@ -25,15 +83,777 @@ const ActorInit En_Kanban_InitVars = {
...
+                            this->direction = -1;
+                        }
+                        this->airTimer = 70;
+                    }
+                    bomb = bomb->next;
+                }
+            }
+            osSyncPrintf("\x1b[32m");
+            osSyncPrintf("OCARINA_MODE %d\n", globalCtx->msgCtx.unk_E3EE);
+            osSyncPrintf("\x1b[m");
```
Comment (fig02) [inline src/overlays/actors/ovl_En_Kanban/z_en_kanban.c:690]: vt

**Review by fig02 — COMMENTED** [review]
thats quite an actor lmfao

File: src/overlays/actors/ovl_En_Kanban/z_en_kanban.c (hunk 1 above)
Comment (petrie911) [reply src/overlays/actors/ovl_En_Kanban/z_en_kanban.c:371]: Surprisingly it doesn't match as a ternary.

**Review by fig02 — APPROVED** [review]

File: src/overlays/actors/ovl_En_Kanban/z_en_kanban_gfx.c [hunk 4]
```diff
@@ -0,0 +1,105 @@
+#include "z_en_kanban.h"
+
+static Gfx* sDisplayLists[] = {
```
[the remark this answers is not in the document]
Comment (petrie911) [reply src/overlays/actors/ovl_En_Kanban/z_en_kanban_gfx.c:3]: I'm including it mostly for consistency. sCutAngles also doesn't really belong in the separate C file but there's no way to get it out.

**Review by Roman971 — APPROVED** [review]

**Pull request merged**